client/src/_transport_layer.rs

```diff
@@ -0,0 +1,257 @@
+        self.ingress_channel.tx.send(forward).ok();
+    }
+
+    async fn spawn_ingress_handler(&self) -> anyhow::Result<()> {
+        let ingress_rx = self
+            .session_layer
+            .receiver()
+            .ok_or_else(|| anyhow::anyhow!("Ingress traffic receiver already spawned"))?;
+
+        tokio::task::spawn_local(self.clone().ingress_handler(ingress_rx));
+        Ok(())
+    }
+
+    async fn ingress_handler(self, ingress_rx: ForwardReceiver) {
+        UnboundedReceiverStream::new(ingress_rx)
+            .for_each(move |forwarded| {
+                let myself = self.clone();
+                async move {
+                    myself.dispatch(forwarded).await;
+                }
+            })
+            .await
+    }
+
+    async fn forward_channel(
+        &self,
+        node_id: NodeId,
+        channel: TransportType,
+    ) -> Option<ForwardSender> {
+        let state = self.state.read().await;
+        match channel {
+            TransportType::Reliable => state.forward_reliable.get(&node_id).cloned(),
+            TransportType::Transfer => state.forward_transfer.get(&node_id).cloned(),
+            TransportType::Unreliable => state.forward_unreliable.get(&node_id).cloned(),
+        }
+    }
+
+    async fn set_forward_channel(
+        &self,
+        node_id: NodeId,
+        channel: TransportType,
+        tx: ForwardSender,
+    ) {
+        let mut state = self.state.write().await;
+        match channel {
+            TransportType::Reliable => state.forward_reliable.insert(node_id, tx),
+            TransportType::Transfer => state.forward_transfer.insert(node_id, tx),
+            TransportType::Unreliable => state.forward_unreliable.insert(node_id, tx),
+        };
+    }
+
+    pub async fn forward(&self, node_id: NodeId) -> anyhow::Result<ForwardSender> {
+        self.forward_generic(node_id, TransportType::Reliable).await
+    }
+
+    pub async fn forward_transfer(&self, node_id: NodeId) -> anyhow::Result<ForwardSender> {
+        self.forward_generic(node_id, TransportType::Transfer).await
+    }
+
+    /// NodeId can be either default or secondary.
+    /// TODO: Make this function resistant to dropping future
+    pub async fn forward_generic(
+        &self,
+        node_id: NodeId,
+        channel: TransportType,
+    ) -> anyhow::Result<ForwardSender> {
+        match self.forward_channel(node_id, channel).await {
+            // If connection was closed in the meantime, it will be initialized on demand.
+            // It will be problematic in some cases, because this can last up to a few seconds.
+            // In worst case scenario initialization will fail and we will wait 5s until timeout.
+            // Since user uses channel, sending will return immediately after item will be taken from
+            // queue, so he won't find out, but the response he expects won't come.
+            // This is argument for changing channels API to `TcpSender`.
```

Comment:
What does it mean that it will be initialized on demand? 
Here we return ForwardSender if it exists so I guess we don't cover that case when connection is closed in the meantime

Reply:
`ForwardSender` has ability to initialize connection on demand (lazy initialization when you send packets).
So we cover this case, because of `ForwardSender::send` implementation.